src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimateId.groovy

```diff
@@ -0,0 +1,59 @@
+package life.qbic.datamodel.dtos.business
+
+/**
+ * A DTO to describe the cost estimate identifier
+ *
+ * A cost estimate is a {@link TomatoId} and describes how the identifier of a cost estimate looks like. It helps to
+ * differentiate between cost estimates
+ *
+ * @since: 1.11.0
+ *
+ */
+class CostEstimateId extends TomatoId{
+    /**
+     * A cost estimate is identified by the type CE
+     */
+    private final String type = "CE"
+    /**
+     * Conserved part of a project
+     */
+    private String projectConserved
+    /**
+     * Random part of the identifier
+     */
+    private String random
+    /**
+     * Version of the identifier
+     */
+    private int version
+
+
+    CostEstimateId(String projectConservedPart, String randomPart, int version) {
+        super(type,projectConservedPart,randomPart,version)
+
+        this.projectConserved = projectConservedPart
+        this.random = randomPart
+        this.version = version
+    }
```

Comment:
`type` is not known in this scope
```suggestion
    CostEstimateId(String projectConservedPart, String randomPart, int version) {
        String type = "CE"
        super(type,projectConservedPart,randomPart,version)
    }    
    private CostEstimateId(String type, String projectConservedPart, String randomPart, int version) {
        super(type,projectConservedPart,randomPart,version)
        this.type = type
        this.projectConserved = projectConservedPart
        this.random = randomPart
        this.version = version
    }
```

Reply:
I would opt for keeping it as an attribute since otherwise, we could also create a QuotationId by just entering Q as type.
I see the type as a characteristic of the class. Why is type not known in this space?

Reply:
I am not against keeping it. It simply does not compile this way

Reply:
Fixed :) 
e71ae379f